### ISSUE TITLE
Error on import: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: Name or service not known

### ISSUE THREAD

**Issue by sammarcus** [Agence-DnD/PIMGento#139]
Similar to #8 and #34 I cannot get past this error:

```SQL
SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: Name or service not known
```
![image](https://cloud.githubusercontent.com/assets/8689264/14062147/ffdeb0bc-f36a-11e5-942b-c6c56bc62a57.png)

Here are the permissions for my root@localhost user

```SQL
GRANT ALL PRIVILEGES ON *.* TO 'root'@'localhost' IDENTIFIED BY PASSWORD '*****' WITH GRANT OPTION 
GRANT PROXY ON ''@'' TO 'root'@'localhost' WITH GRANT OPTION 
```
I am on a Bitnami configuration, here are the permissions for bn_magento@localhost

```SQL
GRANT FILE ON *.* TO 'bn_magento'@'localhost' IDENTIFIED BY PASSWORD '*****'
GRANT ALL PRIVILEGES ON `bitnami_magento`.* TO 'bn_magento'@'localhost'
```

The solution on #8 and #34 (and even others, like #107,) all suggest to grant file permissions to the mysql user, which I believe I have done, and it is reflected when executing `SHOW GRANTS`.
When I execute this query it shows zero rows affected:

```SQL
mysql> GRANT FILE ON *.* TO 'bn_magento'@'%';
Query OK, 0 rows affected (0.00 sec)
```

Even though the error suggests a name resolution issue, bitnami sets the database up on the same instance as the application itself, and my hosts file matches `localhost` to `127.0.0.1`. Of course, I can ping both without issue.

If the pimgento error is a permissions issue, the wording of the error message is deceiving but I am not sure what other permissions to grant. If the error is infact around host name resolution, I'm not sure what how I can rectify it as every connection seems to be fully operational. 

I've also confirmed `local_infile` is enabled:

```SQL
mysql> SHOW GLOBAL VARIABLES LIKE 'local_infile';       
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| local_infile  | ON    |
+---------------+-------+
```

Here is my host node from my local.xml, as you requested in the other issue:
` <host><![CDATA[/opt/bitnami/mysql/tmp/mysql.sock]]></host>`

Lastly, here are my socket configurations:

```SQL
+-----------------------------------------+-----------------------------------+
| Variable_name                           | Value                             |
+-----------------------------------------+-----------------------------------+
| performance_schema_max_socket_classes   | 10                                |
| performance_schema_max_socket_instances | 322                               |
| socket                                  | /opt/bitnami/mysql/tmp/mysql.sock |
+-----------------------------------------+-----------------------------------+
```

Do you have any suggestions how I can solve this?
Thank you!

**Comment by sammarcus** [comment]
+ @DnD-Barracuda @DnD-Fute @DnD-Grifter @DnD-Mimosa @magentix  I'm sorry to bother you, we've tried contacting DnD via the contact form and have not heard back. I understand this is open source but would gladly pay for support if this issue can be solved. Thank you for your time!

**Comment by buttash** [comment]
Hi Sam
DnD are obviously very busy as they don't seem to be able to find time to respond to these issues.

I'm no expert on this, but I think the problem looks similar to the problem I have with apparmor which is an application that restricts application access to the volume.  I found that by adding correct permissions to mysql folders (default location var/lib/mysql and sub-folders but varies when I deploy using Vagrant) you can get mysql to read files inline.  I found this a useful resource - http://askubuntu.com/questions/137424/moving-mysql-datadir.  Good luck!

Martin

**Comment by linslin** [comment]
https://github.com/Agence-DnD/PIMGento/issues/51
https://github.com/Agence-DnD/PIMGento/issues/61

**Comment by sammarcus** [comment]
In the interest of completeness this was solved by disabling "Use MySql Load Data File". I am leaving this issue open because 

1) the error message does not describe the actual problem (in fact, it suggests something completely different all together)
2) after checking with other users it seems this should really be disabled by everyone by default after initial install of PIMGento.

I do not think it is appropriate for DnD to completely ignore issues and am only stating that here to set the expectations for future contributors to not rely on a response (that you will likely not be receiving).

**Comment by buttash** [comment]
I'm aware of a number of (adverse) comments regarding the support of this plugin.  To be fair to Agence-DnD (and I have no association or contact with them), they have put out a free Magento extension that does allow the (basic) connection of Akeneo the Magento, and for that I am am grateful.  Without it I would not be able to use Akeneo on my store.

I think the real solution here is for Akeneo to recognise the importance of a Magento integration to their platform and to take this project on in-house in much the same way as they support other Akeneo add-ons.

Martin

**Comment by sammarcus** [comment]
@buttash I neglected to thank you for your initial suggestion. Any help is much appreciated and I am grateful you lent a hand. 

You might be interested to learn that Akeneo did _used_ to make their in house connector: https://github.com/akeneo/MagentoConnectorBundle but it was deprecated when they decided to formally adopt PIMGento. More of the business decision behind this is outlined in Akeneo's blog post [here](https://www.akeneo.com/blog/goodbye-magentoconnectorbundle-hello-pimgento/) (written exactly one year ago today!).

I have also passed on feedback to Akeneo about my concern with DnD. I understand it is open source but after not getting help here I did contact DnD through their form on their site pleading to purchase support tickets or consulting hours and they also flat out ignored me. It would be a dream if the Akeneo Magento Connector was revived since Akeneo staff is so responsive and pleasant to work with. I cannot use a PIM without some sort of connector between that PIM and Magento, unfortunately making Akeneo impractical for my stack.
